mongoengine/base/datastructures.py

```diff
@@ -187,6 +188,172 @@ def _mark_as_changed(self, key=None):
+
+        Raises ``DoesNotExist`` if the arguments used to query an embedded
+        document returns no results. ``MultipleObjectsReturned`` if more
+        than one result is returned.
+        """
+        values = self.__only_matches(self, kwargs)
+        if len(values) == 0:
+            raise DoesNotExist(
+                "%s matching query does not exist." % self._name
+            )
+        elif len(values) > 1:
+            raise MultipleObjectsReturned(
+                "%d items returned, instead of 1" % len(values)
+            )
+
+        return values[0]
+
+    def first(self):
+        """
+        Returns the first embedded document in the list, or ``None`` if empty.
+        """
+        if len(self) > 0:
+            return self[0]
+
+    def create(self, **values):
+        """
+        Creates a new embedded document and saves it to the database.
+
+        .. note::
+            The entire ancestor collection is saved, any other changes
+            made to the collection will also be saved.
+
+        :param values: A dictionary of values for the embedded document.
+        :return: The new embedded document instance.
+        """
+        instance = self._instance
+        EmbeddedClass = instance._fields[self._name].field.document_type_obj
+        self._instance[self._name].append(EmbeddedClass(**values))
+        instance.save()
+
+        return self._instance[self._name][-1]
+
+    def save(self, *args, **kwargs):
+        """
+        Saves the ancestor collection.
+
+        :param args: Arguments passed up to ancestor collection's Document save
+         method.
+        :param kwargs: Keyword arguments passed up to the ancestor collection's
+         save method.
+        """
+        self._instance.save(*args, **kwargs)
+
+    def delete(self):
+        """
+        Deletes the embedded documents from the database.
+
+        .. note::
+            The entire ancestor collection is saved, any other changes
+            made to the collection will also be saved.
+
+        :return: The number of entries deleted.
+        """
+        values = list(self)
+        for item in values:
+            self._instance[self._name].remove(item)
+        self._instance.save()
+        self._instance.reload()
+
+        return len(values)
+
+    def update(self, **update):
+        """
+        Updates the embedded documents with the given update values.
+
+        .. note::
+            The entire ancestor collection is saved, any other changes
+            made to the collection will also be saved.
+
+        :param update: A dictionary of update values to apply to each
+         embedded document.
+        :return: The number of entries updated.
+        """
+        if len(update) == 0:
+            return 0
+        values = list(self)
+        for item in values:
+            for k, v in update.items():
+                setattr(item, k, v)
+        self._instance.save()
```

Comment:
The idea is to mimic the behavior of a queryset, whose `update()` method changes the values immediately.